cla_backend/apps/reports/forms.py

```diff
@@ -298,6 +298,106 @@ def get_queryset(self):
         return self.execute_query(sql, sql_args)
 
 
+class MICaseExtractExtended(SQLFileDateRangeReport):
+    QUERY_FILE = "MIExtractByOutcomeExtended.sql"
+
+    passphrase = forms.CharField(
+        required=False, help_text="Optional. If not provided, the report will not include diversity data"
+    )
+
+    def get_headers(self):
+        return [
+            "LAA_Reference",
+            "Hash_ID",
+            "Case_ID",
+            "Split_Check",
+            "Split_Link_Case",
+            "Provider_ID",
+            "Category_Name",
+            "Date_Case_Created",
+            "Last_Modified_Date",
+            "Outcome_Code_Child",
+            "Billable_Time",
+            "Cumulative_Time",
+            "Matter_Type_1",
+            "Matter_Type_2",
+            "User_ID",
+            "Scope_Status",
+            "Eligibility_Status",
+            "Adjustments_BSL",
+            "Adjustments_LLI",
+            "Adjustments_MIN",
+            "Adjustments_TYP",
+            "Adjustments_CallbackPreferred",
+            "Adjustments_Skype",
+            "Gender",
+            "Ethnicity",
+            "Age(Range)",
+            "Religion",
+            "Sexual_Orientation",
+            "Disability",
+            "Time_of_Day",
+            "Reject_Reason",
+            "Media_Code",
+            "Contact_Type",
+            "Call_Back_Request_Time",
+            "Call_Back_Actioned_Time",
+            "Time_to_OS_Access",
+            "Time_to_SP_Access",
+            "Residency_Test",
+            "Repeat_Contact",
+            "Referral_Agencies",
+            "Complaint_Type",
+            "Complaint_Date",
+            "Complaint_Owner",
+            "Complaint_Target",
+            "Complaint_Subject",
+            "Complaint_Classification",
+            "Complaint_Outcome",
+            "Agree_Feedback",
+            "Exempt_Client",
+            "Welsh",
+            "Language",
+            "Outcome_Created_At",
+            "Username",
+            "Has_Third_Party",
+            "Time_to_OS_Action",
+            "Organisation",
+            "Notes",
+            "Provider Notes",
+        ]
+
+    def get_rows(self):
+        for row in self.get_queryset():
+            full_row = list(row)
+            diversity_json = full_row.pop() or {}
+
+            def insert_value(key, val):
+                index = self.get_headers().index(key)
+                full_row.insert(index, val)
+
+            insert_value("Gender", diversity_json.get("gender"))
```

Comment:
@Pheonnexx Diversity fields can be return in the sql
https://github.com/ministryofjustice/cla_backend/blob/master/cla_backend/apps/reports/sql/obiee/export_legal_aid_personaldetails.sql#L22

Reply:
This is the same code as the original Case Extract so reluctant to change this for now.